app/models/course/assessment/answer.rb

```diff
@@ -39,10 +39,9 @@ def auto_grade!
     fail ArgumentError unless question.auto_gradable?
     fail IllegalStateError if attempting?
 
-    self.class.transaction do
-      ensure_auto_grading!
-      create_and_queue_auto_grading
-    end
+    auto_grading_job = Course::Assessment::Answer::AutoGradingJob.
```

Comment:
@lowjoel The auto_grading must be created first, then job can use it.  So that I removed the transaction in this case.